Lib/python/std_filesystem.i

```diff
@@ -0,0 +1,85 @@
+/*
+  filesystem::path
+*/
+%{
+#if __cplusplus >= 201703L
+#include <filesystem>
+#include <Python.h>
+#endif // __cplusplus >= 201703L
+%}
+
+%fragment("common", "header") {
+    PyObject * importPathCls() {
```

Comment:
These helpers should probably all be `SWIGINTERN`, e.g.:

```
SWIGINTERN PyObject * importPathCls() {
```
```

Reply:
Just trying to understand so this sticks for next time... So SWIGINTERN means `static` right? [1] 
So that gives it internal linkage, and could be inlined. And we want static not inline because it's not used outside of this TU. Am I following this correctly?

---

[1] and `__attribute__ ((__unused__))` but let's not care

Reply:
From `Lib/swiglabels.swg`:

```
/* internal SWIG method */
#ifndef SWIGINTERN
# define SWIGINTERN static SWIGUNUSED
#endif
```

So it takes care of the adding an `unused` attribute for you (at least for compilers where SWIG knows how to, but it should for ones where `__attribute__ ((__unused__))` works.

The key thing is we don't want these functions to be visible outside the shared object as they aren't needed there.
The load time for a shared object depends on the number of external symbols (as they need relocating).  There's also a small runtime overhead to calling an external symbol even from within the object that defines it since unless you link with special options it's required to be "interposable" (this is how LD_PRELOAD libraries such as libeatmydata work).

Also we'd then have to worry about what happens when more than one SWIG-wrapped module gets loaded, and potentially calls to these functions from one module get routed to the definition in another, which means updating the code of these functions would need extra care to maintain compatibility with how older generated code used them.

I suggested `SWIGINTERN` rather than `SWIGINTERNINLINE` because if we're going to suggest the compiler inline these then I'm unclear why we've gone to the trouble of pulling them out into a helper functions rather than just hand inlining the code into the typemap (which would make avoiding that leak much simpler).

A side issue (sorry if I noted this already): these helpers should really be named such that they're not going to collide with code being wrapped (e.g. consider what happens if the library I'm wrapping has a `pathToStr` function).  We generally use a `SWIG_` prefix, but it's probably better to make that `SWIG_std_filesytem_` here (e.g. `SWIG_std_filesystem_pathToStr`) so we don't have to worry about it potentially colliding with helpers from other SWIG-supplied macros either.

Reply:
Thanks for the explanation, this is helpful.
I have renamed the fragments, I didn't know I needed to also rename the helpers within the fragments themselves.